packages/react/src/Accordion/Accordion.test.tsx

```diff
@@ -0,0 +1,535 @@
+        <Accordion.Content>Test content</Accordion.Content>
+      </Accordion>,
+    )
+
+    const heading = getByRole('heading', {name: 'Test heading'})
+
+    await user.click(heading)
+    expect(onToggle).toHaveBeenCalledTimes(1)
+
+    await user.click(heading)
+    expect(onToggle).toHaveBeenCalledTimes(2)
+  })
+
+  it('calls onKeyDown when a key is pressed', async () => {
+    const onKeyDown = jest.fn()
+    const {getByRole} = render(
+      <Accordion onKeyDown={onKeyDown}>
+        <Accordion.Heading>Test heading</Accordion.Heading>
+        <Accordion.Content>Test content</Accordion.Content>
+      </Accordion>,
+    )
+
+    const details = getByRole('group')
+
+    // Using fireEvent as userEvent doesn't trigger the keydown callback
+    fireEvent.keyDown(details)
+    expect(onKeyDown).toHaveBeenCalledTimes(1)
+
+    fireEvent.keyDown(details)
+    expect(onKeyDown).toHaveBeenCalledTimes(2)
+  })
+
+  it('calls handleOpen with correct value when opened and closed', async () => {
+    const user = userEvent.setup()
+    const handleOpen = jest.fn()
+    const {getByRole} = render(
+      <Accordion handleOpen={handleOpen}>
+        <Accordion.Heading>Test heading</Accordion.Heading>
+        <Accordion.Content>Test content</Accordion.Content>
+      </Accordion>,
+    )
+
+    const heading = getByRole('heading', {name: 'Test heading'})
+
+    await user.click(heading)
+    expect(handleOpen).toHaveBeenCalledWith(true)
+
+    await user.click(heading)
+    expect(handleOpen).toHaveBeenCalledWith(false)
+  })
+
+  it('supports functional ref for Accordion', async () => {
+    const user = userEvent.setup()
+    const functionalRef = jest.fn()
+
+    const {getByRole} = render(
+      <Accordion ref={functionalRef}>
+        <Accordion.Heading>Test heading</Accordion.Heading>
+        <Accordion.Content>Test content</Accordion.Content>
+      </Accordion>,
+    )
+
+    const details = getByRole('group')
+    const heading = getByRole('heading')
+
+    expect(functionalRef).toHaveBeenCalledWith(details)
+
+    expect(details).not.toHaveAttribute('open')
+
+    await user.click(heading)
+    expect(details).toHaveAttribute('open')
+
+    await user.click(heading)
+    expect(details).not.toHaveAttribute('open')
+  })
+
+  it('supports RefObject ref for Accordion', async () => {
+    const user = userEvent.setup()
+    const ref = createRef<HTMLDetailsElement>()
+    const {getByRole} = render(
+      <Accordion ref={ref}>
+        <Accordion.Heading>Test heading</Accordion.Heading>
+        <Accordion.Content>Test content</Accordion.Content>
+      </Accordion>,
+    )
+
+    const details = getByRole('group')
+    const heading = getByRole('heading')
+
+    expect(ref.current).toBe(details)
+
+    expect(details).not.toHaveAttribute('open')
+
+    await user.click(heading)
+    expect(details).toHaveAttribute('open')
+
+    await user.click(heading)
+    expect(details).not.toHaveAttribute('open')
```

Comment:
Is the reason for adding these extra assertions that you're checking the passing of ref doesn't cause open/close regressions? Was wondering how they're related to testing the ref.